app/test_util.py

```diff
@@ -37,22 +38,57 @@ def get(self, uri):
 
     def wait_for_xpath(self, xpath, timeout=5):
         return WebDriverWait(self, timeout).until(
-            expected_conditions.presence_of_element_located((By.XPATH, xpath)))
+            expected_conditions.presence_of_element_located((By.XPATH, xpath))
+        )
+
+    def wait_for_css(self, css, timeout=5):
+        return WebDriverWait(self, timeout).until(
+            expected_conditions.presence_of_element_located((By.CSS, css))
+        )
 
     def wait_for_xpath_to_disappear(self, xpath, timeout=5):
         return WebDriverWait(self, timeout).until_not(
-            expected_conditions.presence_of_element_located((By.XPATH, xpath)))
+            expected_conditions.presence_of_element_located((By.XPATH, xpath))
+        )
+
+    def wait_for_css_to_disappear(self, css, timeout=5):
+        return WebDriverWait(self, timeout).until_not(
+            expected_conditions.presence_of_element_located((By.CSS, css))
+        )
 
     def wait_for_xpath_to_be_clickable(self, xpath, timeout=5):
         return WebDriverWait(self, timeout).until(
-            expected_conditions.element_to_be_clickable((By.XPATH, xpath)))
+            expected_conditions.element_to_be_clickable((By.XPATH, xpath))
+        )
+
+    def wait_for_css_to_be_clickable(self, css, timeout=5):
+        return WebDriverWait(self, timeout).until(
+            expected_conditions.element_to_be_clickable((By.CSS, css))
+        )
 
     def scroll_to_element_and_click(self, element):
-        try:
-            return element.click()
-        except ElementClickInterceptedException:
-            self.execute_script("arguments[0].scrollIntoView();", element)
-            return element.click()
+        ActionChains(self).move_to_element(element).perform()
+        return element.click()
+
+    def click_xpath(self, xpath):
+        element = self.wait_for_xpath_to_be_clickable(xpath)
+        return self.scroll_to_element_and_click(element)
+
+    def click_css(self, css):
+        element = self.wait_for_css_to_be_clickable(css)
+        return self.scroll_to_element_and_click(element)
+
+    def become_user(self, user_id):
+        self.wait_for_css("body").send_keys(Keys.CONTROL + "t")
+        self.get(f'/become_user/{user_id}')
+        self.wait_for_css("body").send_keys(Keys.CONTROL + "w")
+
+        self.execute_script(f"window.open('{self.server_url + '/become_user/{user_id}'}', 'new_window')")
```

Comment:
Didn't you just do this above on line 83? Or can you help explain what's going on here if I'm just not seeing it?

Reply:
Shoot, how did that end up in there; I'm sorry, I'll take it out again.